File: LC75/1768. Merge Strings Alternately/MSA.java
public class MSA {
    public String mergeAlternately(String word1, String word2) {
        StringBuilder mergedString = new StringBuilder();
        int i = 0, j = 0;
        
        while (i < word1.length() && j < word2.length()) {
            mergedString.append(word1.charAt(i));
            mergedString.append(word2.charAt(j));
            i++;
            j++;
        }
        
        if (i < word1.length()) {
            mergedString.append(word1.substring(i));
        }
        if (j < word2.length()) {
            mergedString.append(word2.substring(j));
        }
        
        return mergedString.toString();
    }
}

File: LC75/1768. Merge Strings Alternately/MSA.ts
function mergeAlternately(word1: string, word2: string): string {
    let mergedString = '';
        const maxLength = Math.max(word1.length, word2.length);

        for (let i = 0; i < maxLength; i++) {
            if (i < word1.length) {
                mergedString += word1[i];
            }
            if (i < word2.length) {
                mergedString += word2[i];
            }
        }

        return mergedString;
};